apps/ima/behaviours/save-form-session.js

```diff
@@ -75,15 +75,33 @@ module.exports = superclass => class extends superclass {
 
         const isContinueOnEdit = req.form.options.continueOnEdit &&
           _.get(req.form.options.forks, '[0].continueOnEdit');
-        // Redirect to final-summary if on the edit flow of the harm-claim page
-        if((req.sessionModel.get('steps')[req.sessionModel.get('steps').length - 1] === '/harm-claim-summary'
-        || req.sessionModel.get('steps')[req.sessionModel.get('steps').length - 1] === '/harm-claim' ||
-        req.sessionModel.get('steps')[req.sessionModel.get('steps').length - 1] === '/harm-claim-countries' ||
-        req.sessionModel.get('steps')[req.sessionModel.get('steps').length - 1].includes('/risk-of-harm') ||
-        req.sessionModel.get('steps')[req.sessionModel.get('steps').length - 1].includes('/harm-claim-details')) &&
-        (req.form.options.continueOnEdit === false || req.sessionModel.get('is-serious-and-irreversible') === 'no') &&
-        req.sessionModel.get('steps').includes('/human-rights-claim')) {
-          return res.redirect('/ima/final-summary');
+
+        const sessionSteps = req.sessionModel.get('steps');
+        const formSteps = Object.keys(req.form.options.steps);
+        const orderedSessionSteps = formSteps.filter(step => sessionSteps.includes(step));
+        const latestStepInJourney = orderedSessionSteps[orderedSessionSteps.length - 1];
+
+        // logic to override the SIH routing to resolve issues with routing
+        if(JSON.stringify(req.form.options.route, null, 2) === '"/harm-claim"' &&
```

Comment:
Does it need to be stringified? Could it not just be ` if(req.form.options.route) === '/harm-claim'`?

Reply:
Tested without the stringify() function, but it did not pick up the value.

Reply:
Could that be because you have /harm-claim in two sets of quotes - single and double` '" "'`?

Reply:
Yeah, re-tested with just single quotes and is working fine, thank you. With the JSON.stringify() it was returning the value with double quotes.